nixos/modules/services/security/authentik.nix

```diff
@@ -0,0 +1,207 @@
+    mkPackageOption
+    ;
+  cfg = config.services.authentik;
+
+  isRedisUnixSocket = lib.hasPrefix "/" cfg.redis.host;
+
+  commonServiceConfig = {
+    Type = "simple";
+    User = "authentik";
+    Group = "authentik";
+    Restart = "always";
+    RestartSec = 3;
+    RuntimeDirectory = "authentik";
+    NoNewPrivileges = true;
+    PrivateTmp = true;
+    ProtectHome = true;
+    ProtectSystem = "strict";
+    ProtectKernelTunables = true;
+    ProtectKernelModules = true;
+    ProtectControlGroups = true;
+    SystemCallFilter = "~@cpu-emulation @keyring @module @obsolete @raw-io @reboot @swap @sync";
+    ConfigurationDirectory = "authentik";
+  };
+in
+{
+  options.services.authentik = {
+    enable = mkEnableOption "Authentik";
+    package = mkPackageOption pkgs "authentik" { };
+
+    secretsFile = mkOption {
+      type = types.str // {
+        # We don't want users to be able to pass a path literal here but
+        # it should look like a path.
+        check = it: lib.isString it && lib.types.path.check it;
+      };
+      example = "/run/secrets/authentik";
+      description = ''
+        Path of a file with extra environment variables to be loaded from disk. This file is not added to the nix store, so it can be used to pass secrets to authentik. Refer to the [documentation](https://docs.goauthentik.io/docs/installation/configuration) for options.
+
+        Authentik requires at least a secret key:
+        ```
+        AUTHENTIK_SECRET_KEY=<random key>
+        ```
+      '';
+    };
+    environment = mkOption {
+      type = types.submodule { freeformType = types.attrsOf types.str; };
+      default = { };
+      example = {
+        AUTHENTIK_LOG_LEVEL = "debug";
+      };
+      description = ''
+        Extra configuration environment variables. Refer to the [documentation](https://docs.goauthentik.io/docs/installation/configuration) for options.
+      '';
+    };
+
+    host = lib.mkOption {
+      type = types.str;
+      default = "localhost";
+      description = "The host that authentik will listen on.";
+    };
+    port = mkOption {
+      type = types.port;
+      default = 9000;
+      description = "The port that immich will listen on.";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The user immich should run as.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The group immich should run as.";
+    };
+    database = {
+      enable =
+        mkEnableOption "the postgresql database for use with authentik. See {option}`services.postgresql`"
+        // {
+          default = true;
+        };
+      createDB = mkEnableOption "the automatic creation of the database for authentik." // {
+        default = true;
+      };
+      name = mkOption {
+        type = types.str;
+        default = "authentik";
+        description = "The name of the authentik database.";
+      };
```

Comment:
Add an assert that `database.name == user` if `database.createDB` is true.

Reply:
Why?

Reply:
Because otherwise the Postgres module is going to complain about not having matching user and database names, so having the authentik module itself complain would make it more explicit to the user what the issue is.